### PR TITLE
[13.0][MIG] migration of module stock_financial_risk

### PR DESCRIPTION
**Pull request by alan196**
### Proposed changes

I've migrated the module stock_financial_risk and its functionalities from Odoo 12.0 to 13.0. I've checked if all unit tests are still running.

### Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Migration Update

### Checklist

- [x] I have read the CONTRIBUTING doc
- [x] I have signed the CLA
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated necessary documentation

### Further comments

This module depends on the account_financial_risk. That means that there should be a migration of it to 13.0. Because there is already a PR for migration I've added the changes of the PR to my local environment to test my migration.

This PR depends on the current migration PR:

https://github.com/OCA/credit-control/pull/52

### CODE REVIEW

**Comment by pedrobaeza** [comment]
@alan196 please rebase over current branch that has the dependent module migrated.

**Comment by alan196** [comment]
@pedrobaeza rebase done

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
@carlosdauden TT26030

File: stock_financial_risk/models/stock.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    def _action_done(self, cancel_backorder=False):
+        if not self.env.context.get("bypass_risk"):
+            moves = self.filtered(
+                lambda x: (
+                    x.location_dest_id.usage == "customer"
+                    and x.partner_id.risk_exception
```
Comment (carlosdauden) [inline stock_financial_risk/models/stock.py]:
```suggestion
                    and x.partner_id.commercial_partner_id.risk_exception
```

File: stock_financial_risk/models/stock.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                }
+            )
+            .action_show()
+        )
+
+    def action_confirm(self):
+        if not self.env.context.get("bypass_risk"):
+            if (
+                self.location_dest_id.usage == "customer"
+                and self.partner_id.risk_exception
```
Comment (carlosdauden) [inline stock_financial_risk/models/stock.py]:
```suggestion
                and self.partner_id.commercial_partner_id.risk_exception
```

File: stock_financial_risk/models/stock.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            ):
+                return self.show_risk_wizard("action_confirm")
+        return super(StockPicking, self).action_confirm()
+
+    def action_assign(self):
+        if not self.env.context.get("bypass_risk") and self.filtered(
+            "partner_id.risk_exception"
```
Comment (carlosdauden) [inline stock_financial_risk/models/stock.py]:
```suggestion
            "partner_id.commercial_partner_id.risk_exception"
```

File: stock_financial_risk/models/stock.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        ):
+            return self.show_risk_wizard("action_assign")
+        return super(StockPicking, self).action_assign()
+
+    def button_validate(self):
+        if not self.env.context.get("bypass_risk"):
+            if (
+                self.location_dest_id.usage == "customer"
+                and self.partner_id.risk_exception
```
Comment (carlosdauden) [inline stock_financial_risk/models/stock.py]:
```suggestion
                and self.partner_id.commercial_partner_id.risk_exception
```

**Review by carlosdauden — CHANGES_REQUESTED** [review]
In this version we have changed the behavior a bit, so I have suggested the changes to control the risk of commercial_partner_id

**Comment by alan196** [comment]
@carlosdauden changes applied 👍

**Review by carlosdauden — APPROVED** [review]

**Pull request merged**